pinot-controller/src/main/java/com/linkedin/pinot/controller/api/resources/PinotTableConfigRestletResource.java

```diff
@@ -0,0 +1,188 @@
+    CombinedConfig config = null;
+
+    try {
+      config = Deserializer.deserializeFromString(CombinedConfig.class, tableConfiguration);
+    } catch (Exception e) {
+      LOGGER.warn("Caught exception while deserializing the table configuration", e);
+      return Response
+          .serverError()
+          .entity(e.getMessage())
+          .type(MediaType.TEXT_PLAIN_TYPE)
+          .build();
+    }
+
+    if (config == null) {
+      LOGGER.warn("Failed to deserialize the table configuration: {}", tableConfiguration);
+      return Response
+          .serverError()
+          .entity("Failed to deserialize the table configuration")
+          .type(MediaType.TEXT_PLAIN_TYPE)
+          .build();
+    }
+
+    if (config.getSchema() != null) {
+      _resourceManager.addOrUpdateSchema(config.getSchema());
+    }
+
+    if (config.getOfflineTableConfig() != null) {
+      _resourceManager.addTable(config.getOfflineTableConfig());
+    }
+
+    if (config.getRealtimeTableConfig() != null) {
+      _resourceManager.addTable(config.getRealtimeTableConfig());
+    }
+
+    return Response
+        .ok()
+        .build();
+  }
+
+  @PUT
+  @Produces(MediaType.APPLICATION_JSON)
+  @Path("/v2/tables/{tableName}")
+  public Response updateTable(String tableConfiguration) {
+    CombinedConfig config = null;
+
+    try {
+      config = Deserializer.deserializeFromString(CombinedConfig.class, tableConfiguration);
+    } catch (Exception e) {
+      LOGGER.warn("Caught exception while deserializing the table configuration", e);
+      return Response
+          .serverError()
+          .entity(e.getMessage())
+          .type(MediaType.TEXT_PLAIN_TYPE)
+          .build();
+    }
+
+    if (config == null) {
+      LOGGER.warn("Failed to deserialize the table configuration: {}", tableConfiguration);
+      return Response
+          .serverError()
+          .entity("Failed to deserialize the table configuration")
+          .type(MediaType.TEXT_PLAIN_TYPE)
+          .build();
+    }
+
+    if (config.getSchema() != null) {
+      _resourceManager.addOrUpdateSchema(config.getSchema());
+    }
+
+    if (config.getOfflineTableConfig() != null) {
+      if (_resourceManager.getAllTables().contains(config.getOfflineTableConfig().getTableName())) {
+        try {
+          _resourceManager.setExistingTableConfig(config.getOfflineTableConfig(), config.getOfflineTableConfig().getTableName(),
+              CommonConstants.Helix.TableType.OFFLINE);
+        } catch (IOException e) {
+          LOGGER.warn("Failed to update the offline table configuration for table {}", e, config.getOfflineTableConfig().getTableName());
+          return Response
+              .serverError()
+              .entity("Failed to update the offline table configuration")
+              .type(MediaType.TEXT_PLAIN_TYPE)
+              .build();
+        }
+      } else {
+        _resourceManager.addTable(config.getOfflineTableConfig());
```

Comment:
This seems unexpected as this update API. Can we log a message saying we are adding a table?

Reply:
`addTable` in `PinotHelixResourceManager` already logs the table addition (see https://github.com/linkedin/pinot/blob/6fbb79e734a9078190d3f4ee29c62e487c9100dd/pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java#L1120).

The fact that a table is added if a new table counterpart is added to the table configuration is an implementation detail, from the users' perspective, the table types just changed from say `[OFFLINE]` to `[OFFLINE, REALTIME]`.